config/constants.ts

```diff
@@ -24,7 +24,7 @@ const orcaBusStatefulConfig = {
     defaultDatabaseName: 'orcabus',
     version: AuroraPostgresEngineVersion.VER_15_4,
     parameterGroupName: 'default.aurora-postgresql15',
-    username: 'admin',
+    username: 'postgres',
     dbPort: 5432,
     masterSecretName: rdsMasterSecretName,
```

Comment:
We have to create per app/service db secret manager record per database to isolate them better. Instead of reusing this master (SA account). Wonder, we can do this arrangement straight up now at this point... By chance, do you happen to figure any solution to it?

Reply:
I agree that we should isolate on a per service level, but that means the stateless stack needs to be aware of the consuming services, right? 
And any database required by any service would have to be registered to and created by the stateless stack IIRC. The stateless stack can create "user" credentials for each service, but I think they'd need to be linked to existing DBs? I don't have the understanding yet of what's possible...

I think the initial thinking here was to allow access on the cluster level to start with in order to not block any service development. Once we've figured out a better way, we could refactor. 

Or do you think it's worth working that out now and safe on the refactor ?

Reply:
I'd recommend we should work this out now.

Reply:
I agree in working this out. Could be wrong, but it doesn't look like CDK can create more than one database/user per cluster using the regular higher level constructs (see issue https://github.com/aws/aws-cdk/issues/13588)? Might have to create a migration-style Lambda function which executes `create database ...` after the RDS cluster is up and running, e.g. using something like [this](https://github.com/umccr/orcabus/blob/50a6f662ab8723ad3a71595a6ef3f84a1a70f192/lib/workload/stateful/filemanager/deploy/constructs/cdk_resource_invoke.ts#L44). Then, I think each service would need to register their preferred database name, user and credentials somewhere (in constants.ts?).

Reply:
Or, each service could register their own migration function to create the databases/users? That way the databases wouldn't have to be registered somewhere higher up like in `constants.ts`. Not sure if this makes more sense?

Reply:
Hm, I wonder if the stateless stack could create IAM roles/policies for each service that would allow these services access to only their specific DB(s) on the instance. E.g. passing on the permissions to do what the service needs to do (without control/knowledge of what that may be), as long as it happens in the designated DB. 

I haven't found any examples, but it looks like you can create IAM policies with conditions to restrict access by [rds:DatabaseName](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/security_iam_service-with-iam.html#UsingWithRDS.IAM.Conditions).

Reply:
Will track this at issue #99